plugins/vpp/ifplugin/interface_state.go

```diff
@@ -22,46 +22,46 @@ import (
 	"sync"
 	"time"
 
+	"git.fd.io/govpp.git/adapter"
+
 	govppapi "git.fd.io/govpp.git/api"
 	"github.com/go-errors/errors"
 	"github.com/ligato/cn-infra/logging"
 	"github.com/ligato/cn-infra/utils/safeclose"
 	"github.com/ligato/vpp-agent/plugins/govppmux"
 	"github.com/ligato/vpp-agent/plugins/vpp/binapi/interfaces"
-	"github.com/ligato/vpp-agent/plugins/vpp/binapi/stats"
 	"github.com/ligato/vpp-agent/plugins/vpp/ifplugin/ifaceidx"
 	intf "github.com/ligato/vpp-agent/plugins/vpp/model/interfaces"
 )
 
-// counterType is the basic counter type - contains only packet statistics.
-type counterType int
+// PeriodicPollingTimeout between statistics reads
+// TODO set to the original VPP interval but should be configurable
+var PeriodicPollingTimeout = 10 * time.Second
 
-// constants as defined in the vnet_interface_counter_type_t enum in 'vnet/interface.h'
-const (
-	Drop    counterType = 0
-	Punt                = 1
-	IPv4                = 2
-	IPv6                = 3
-	RxNoBuf             = 4
-	RxMiss              = 5
-	RxError             = 6
-	TxError             = 7
-	MPLS                = 8
-)
+// statType is the specific interface statistics type.
+type statType string
 
-// combinedCounterType is the extended counter type - contains both packet and byte statistics.
-type combinedCounterType uint8
+// string used as pattern to filter stats results to interface only
+const ifFilter = "/if"
 
-// constants as defined in the vnet_interface_counter_type_t enum in 'vnet/interface.h'
+// interface statistics matching patterns
 const (
-	Rx          combinedCounterType = 0
-	RxUnicast                       = 1
-	RxMulticast                     = 2
-	RxBroadcast                     = 3
-	Tx                              = 4
-	TxUnicast                       = 5
-	TxMulticast                     = 6
-	TxBroadcast                     = 7
+	Drop        statType = ifFilter + "/drops"
+	Punt                 = ifFilter + "/punt"
+	IPv4                 = ifFilter + "/ip4"
+	IPv6                 = ifFilter + "/ip6"
+	RxNoBuf              = ifFilter + "/rx-no-buf"
+	RxMiss               = ifFilter + "/rx-miss"
+	RxError              = ifFilter + "rx-error"
```

Comment:
Missing slash here. I think it would be better to define these without the slash and add slash to `ifFilter` so it represents prefix.